tools/test/audio/comp_run.sh

```diff
@@ -35,16 +108,30 @@ else
 fi
 
 # Run command
-# There is no more need to specify the library explicitly
-#LIB="-a ${COMP}=libsof_${COMP}.so"
-LIB=""
-ARG="-d -r $FS1 -R $FS2 -i $FN_IN -o $FN_OUT -t $TPLG $BINFMT $LIB"
+OPTS="$DEBUG -r $FS_IN -R $FS_OUT -c $CHANNELS_IN -n $CHANNELS_OUT $BINFMT -t $TPLG"
+DATA="-i $FN_IN -o $FN_OUT"
+ARG="$OPTS $EXTRA_OPTS $DATA"
 CMD="$HOST_EXE $ARG"
+VALGRIND_CMD="valgrind --leak-check=yes --error-exitcode=1"
 export LD_LIBRARY_PATH=$HOST_LIB:$TPLG_LIB
 
 # Run test bench
 echo "Command:         $HOST_EXE"
 echo "Argument:        $ARG"
-echo "LD_LIBRARY_PATH: ${LD_LIBRARY_PATH}"
+echo "LD_LIBRARY_PATH=${LD_LIBRARY_PATH}"
 
-valgrind --leak-check=yes --error-exitcode=1 $CMD
+rm -f "$FN_OUT"
+rm -f "$FN_TRACE"
+if [ "$VALGRIND" = "yes" ]; then
```

Comment:
You don't need the `if VALGRIND` test.

If you drop the `2>` redirection and switch to `--log-file` or something, then you don't need the `if FN_TRACE` either.

You can run this same command with $MAYBE_VALGRIND maybe empty maybe not and it will Just Work:
```
  $MAYBE_VALGRIND $CMD
```
Again `$MAYBE_REDIRECT` does _not_ work, you need the if.

(I'm not suggesting to actually prefix the variable with MAYBE_)

Reply:
LOL -- I might have used maybe :) Good recommendations, thanks!

Reply:
>  I might have used maybe :) 

Maybe it's better :-)

Reply:
Done with $VALGRIND_CMD $CMD, no --log-file yet.